librz/analysis/fcn.c

```diff
@@ -569,6 +586,7 @@ static int run_basic_block_analysis(RzAnalysisTaskItem *item, RzVector *tasks) {
 	bool is_x86 = is_arm ? false : analysis->cur->arch && !strncmp(analysis->cur->arch, "x86", 3);
 	bool is_amd64 = is_x86 ? fcn->cc && !strcmp(fcn->cc, "amd64") : false;
 	bool is_dalvik = is_x86 ? false : analysis->cur->arch && !strncmp(analysis->cur->arch, "dalvik", 6);
+	bool is_hexagon = is_x86 ? false : analysis->cur->arch && !strncmp(analysis->cur->arch, "hexagon", 7);
```

Comment:
these checks are kinda weird, i'm not sure about this, but maybe i would change this to a if-else-if-else statement.